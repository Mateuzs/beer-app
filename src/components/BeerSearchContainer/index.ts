export { default as BeerSearchContainer } from "./BeerSearchContainer";
