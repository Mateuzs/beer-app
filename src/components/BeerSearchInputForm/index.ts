export { default as BeerSearchInputForm } from "./BeerSearchInputForm";
