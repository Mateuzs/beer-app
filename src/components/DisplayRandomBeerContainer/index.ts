export { default as DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
