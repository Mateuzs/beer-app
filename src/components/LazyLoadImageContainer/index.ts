export { default as LazyLoadImageContainer } from "./LazyLoadImageContainer";
