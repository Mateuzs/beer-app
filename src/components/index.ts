export { BeerSearchContainer } from "./BeerSearchContainer";
export { DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
export { NavBar } from "./NavBar";
export { SearchButton } from "./SearchButton";
export { LazyLoadImageContainer } from "./LazyLoadImageContainer";
export { ErrorMessage } from "./ErrorMessage";
export { BeerSearchInputForm } from "./BeerSearchInputForm";
export { BeerList } from "./BeerList";
