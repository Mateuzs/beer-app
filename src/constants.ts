// in this simplified solution all wording has been defined in constants
// in real scenario it's required a translate mechanism to support many languages

export const APP_HEADER = "Beer App";

export const ERROR_MESSAGE = "Something went wrong! Try again later.";

export const LABEL_SEARCH_BEER = "Another beer!";
export const LABEL_SEARCH_NON_ALCOHOLIC_BEER = "Random non alcoholic beer";
export const LOADING_INFO = "Loading...";
export const IMAGE_NOT_AVAILABLE_INFO = "Image not available.";
export const IMAGE_URL_PLACEHOLDER = "Loading...";
export const LAZY_IMAGE_BLUR_EFFECT = "blur";
export const BEER_IMAGE_HEIGHT_PX = 300;
export const SMALL_BEER_IMAGE_HEIGHT_PX = 150;

export const BEER_SEARCH_INPUT_DATE_DESCRIPTION = "format date: MM-YYYY";
export const BEER_SEARCH_INPUT_NAME_DESCRIPTION = "Only letters and numbers";
export const BEER_SEARCH_INPUT_DATE_PLACEHOLDER = "Date";
export const BEER_SEARCH_INPUT_NAME_PLACEHOLDER = "Name";

export const BEER_SEARCH_INPUT_RADIO_NAME = "beer-settings";
export const BEER_SEARCH_INPUT_RADIO_TYPE = "radio";
export const BEER_SEARCH_INPUT_RADIO_NAME_DESCRIPTION = "Name";
export const BEER_SEARCH_INPUT_RADIO_DATE_DESCRIPTION = "Brewed Before";
export const BEER_SEARCH_INFO = "Nothing to display";

export const SEARCH_BUTTON_LABEL = "Search!";
