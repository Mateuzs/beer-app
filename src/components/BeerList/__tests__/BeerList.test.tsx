// external
import React from "react";
import { shallow, mount } from "enzyme";
// components
import { BeerList } from "../../../components";
// constants, utils
import { BEER_SEARCH_INFO } from "../../../constants";

describe("BeerList component", () => {
  test("should match the snapshot and styles", () => {
    const mockedData = {
      isFetchedBeerList: true,
      beerList: [
        {
          id: 1,
          name: "BEER",
          imageUrl: "BEER_URL",
          description: "DESCRIPTION",
        },
      ],
    };
    const wrapper = shallow(<BeerList {...mockedData} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should render list of beers", () => {
    const mockedData = {
      isFetchedBeerList: true,
      beerList: [
        {
          id: 1,
          name: "BEER",
          imageUrl: "BEER_URL",
          description: "DESCRIPTION",
        },
      ],
    };
    const component = shallow(<BeerList {...mockedData} />);

    expect(component.find('[data-test-id="beer-list"]').length).toEqual(1);
    expect(component.find('[data-test-id="beer-search-info"]').length).toEqual(0);
  });

  test("should render proper info when empty list fetched", () => {
    const mockedData = {
      isFetchedBeerList: true,
      beerList: [],
    };
    const component = shallow(<BeerList {...mockedData} />);

    expect(component.find('[data-test-id="beer-list"]').length).toEqual(0);
    expect(component.find('[data-test-id="beer-search-info"]').length).toEqual(1);
    expect(component.find('[data-test-id="beer-search-info"]').text()).toEqual(BEER_SEARCH_INFO);
  });

  test("should render null when no list fetched", () => {
    const mockedData = {
      isFetchedBeerList: false,
      beerList: [],
    };
    const component = shallow(<BeerList {...mockedData} />);

    expect(component.find('[data-test-id="beer-list"]').length).toEqual(0);
    expect(component.find('[data-test-id="beer-search-info"]').length).toEqual(0);
  });
});
