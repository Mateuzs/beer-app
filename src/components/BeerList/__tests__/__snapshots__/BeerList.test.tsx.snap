// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerList component should match the snapshot and styles 1`] = `
<div
  className="beer-list-container"
>
  <ul
    className="beer-list"
    data-test-id="beer-list"
  >
    <li
      className="beer-element"
      key="1"
    >
      <div
        className="beer-element-image"
      >
        <Memo(LazyLoadImageContainer)
          height={150}
          imageUrl="BEER_URL"
        />
      </div>
      <div
        className="beer-element-descripiton"
      >
        <p
          className="beer-element-title"
        >
          BEER
        </p>
        <div
          className="beer-element-description"
        >
          DESCRIPTION
        </div>
      </div>
    </li>
  </ul>
</div>
`;
